ganttproject/src/main/java/net/sourceforge/ganttproject/storage/SqlProjectDatabaseImpl.kt

```diff
@@ -63,23 +66,70 @@ class SqlProjectDatabaseImpl(private val dataSource: DataSource): ProjectDatabas
   }
 
   @Throws(ProjectDatabaseException::class)
-  override fun init(): Unit = withDSL({ "Failed to initialize the database" }) { dsl ->
-    dsl
-      .createTable(TASK)
-      .column(TASK.ID, INTEGER)
-      .column(TASK.NAME, VARCHAR.notNull())
-      .column(TASK.COLOR, VARCHAR.null_())
-      .constraints(primaryKey(TASK.ID))
-      .execute()
+  override fun init() {
+    val scriptStream = javaClass.getResourceAsStream(DB_INIT_SCRIPT_PATH)
+    if (scriptStream == null) {
+      val message = "Init script not found"
+      LOG.error(message)
+      throw ProjectDatabaseException(message)
+    }
+    try {
+      val queries = String(scriptStream.readAllBytes(), Charsets.UTF_8)
+
+      dataSource.connection.use { connection ->
+        connection.createStatement().execute(queries)
+      }
+    } catch (e: Exception) {
+      val message = "Failed to init the database {}"
+      LOG.error(message, e)
+      throw ProjectDatabaseException(message)
+    }
   }
 
   @Throws(ProjectDatabaseException::class)
   override fun insertTask(task: Task): Unit = withDSL({ "Failed to insert task ${task.taskID}" }) { dsl ->
+    val priority = task.priority.let { pr -> if (pr != Task.DEFAULT_PRIORITY) pr.persistentValue else null }
+    val webLink = task.webLink.let { link ->
+      if (!link.isNullOrBlank() && link != "http://") URLEncoder.encode(link, Charsets.UTF_8.name()) else null
+    }
+    var costManualValue: String? = null
+    var isCostCalculated: Boolean? = null
+    if (!(task.cost.isCalculated && task.cost.manualValue == BigDecimal.ZERO)) {
+      costManualValue = task.cost.manualValue.toPlainString()
+      isCostCalculated = task.cost.isCalculated
+    }
+    val notes = task.notes?.let { notes -> notes.replace("\\r\\n", "\\n").ifBlank { null } }
```

Comment:
This was basically copied from TaskSaver, right? Can you please add a TODO to use common serialization code in the future?